sklearn/utils/_tags.py

```diff
@@ -1,71 +1,318 @@
+        Whether to skip common tests entirely. Don't use this unless
+        you have a *very good* reason.
+
+    _xfail_checks : dict[str, str] (default={})
+        Dictionary ``{check_name: reason}`` of common checks that will
+        be marked as `XFAIL` for pytest, when using
+        :func:`~sklearn.utils.estimator_checks.parametrize_with_checks`. These
+        checks will be simply ignored and not run by
+        :func:`~sklearn.utils.estimator_checks.check_estimator`, but a
+        `SkipTestWarning` will be raised.  Don't use this unless there
+        is a *very good* reason for your estimator not to pass the
+        check.  Also note that the usage of this tag is highly subject
+        to change because we are trying to make it more flexible: be
+        prepared for breaking changes in the future.
+
+    input_tags : InputTags
+        The input data(X) tags.
+    """
+
+    target_tags: TargetTags
+    transformer_tags: TransformerTags | None
+    classifier_tags: ClassifierTags | None
+    regressor_tags: RegressorTags | None
+    array_api_support: bool = False
+    no_validation: bool = False
+    non_deterministic: bool = False
+    requires_fit: bool = True
+    _skip_test: bool = False
+    _xfail_checks: dict[str, str] = field(default_factory=dict)
+    input_tags: InputTags = field(default_factory=InputTags)
+
+
+def default_tags(estimator) -> Tags:
+    """Get the default tags for an estimator.
+
+    This ignores any ``__sklearn_tags__`` method that the estimator may have.
+
+    If the estimator is a classifier or a regressor, ``target_tags.required``
+    will be set to ``True``, otherwise it will be set to ``False``.
+
+    ``transformer_tags`` will be set to ``TransformerTags()`` if the estimator
+    has a ``transform`` or ``fit_transform`` method, otherwise it will be set
+    to ``None``.
+
+    ``classifier_tags`` will be set to ``ClassifierTags()`` if the estimator is
+    a classifier, otherwise it will be set to ``None``.
+
+    ``regressor_tags`` will be set to ``RegressorTags()`` if the estimator is a
+    regressor, otherwise it will be set to ``None``.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator for which to get the default tags.
+
+    Returns
+    -------
+    tags : Tags
+        The default tags for the estimator.
+    """
+    from ..base import is_classifier, is_regressor
+
+    target_required = is_classifier(estimator) or is_regressor(estimator)
+
+    return Tags(
+        target_tags=TargetTags(required=target_required),
+        transformer_tags=(
+            TransformerTags()
+            if hasattr(estimator, "transform") or hasattr(estimator, "fit_transform")
+            else None
+        ),
+        classifier_tags=ClassifierTags() if is_classifier(estimator) else None,
+        regressor_tags=RegressorTags() if is_regressor(estimator) else None,
+    )
+
+
+def get_tags(estimator) -> Tags:
+    """Get estimator tags.
```

Comment:
Is there / should there be a corresponding method for each type of tags? 

E.g. `get_input_tags`, etc.? Or am I missing something?

Reply:
No this returns the tag root, and then one can use `tags.input_tags....` to access the inner tags.